pkg/metrics/timer.go

```diff
@@ -0,0 +1,42 @@
+package metrics
+
+import (
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+// Timer is the struct that keeps track of each of the sections.
+type Timer struct {
+	startTime    time.Time
+	totalTime    time.Time
+	listOfStages []string
+	listOfTimes  []time.Duration
+}
+
+var timer Timer
+
+// StartTimer initiailzes the timer object with the current timestamp information.
+func StartTimer() {
+	timer = Timer{}
+	timer.startTime = time.Now().Round(time.Second)
+	timer.totalTime = time.Now().Round(time.Second)
+	timer.listOfStages = []string{"Infrastructure", "API", "Bootstrap Complete", "Bootstrap Destroy", "Cluster operator"}
+}
+
+// SetTimeElapsed returns the duration of the current stage.
+func SetTimeElapsed() {
+	duration := time.Now().Round(time.Second).Sub(timer.startTime)
+	timer.listOfTimes = append(timer.listOfTimes, duration)
+	timer.startTime = time.Now().Round(time.Second)
+}
+
+// LogSummary prints the summary of all the times collected so far into the INFO section.
+func LogSummary() {
+	logrus.Infof("Time elapsed for each section")
+	indent := "	"
+	for index, item := range timer.listOfStages {
+		logrus.Infof(indent+"%s : %v", item, timer.listOfTimes[index])
+	}
+	logrus.Infof("Total time elapsed : %v", time.Now().Round(time.Second).Sub(timer.totalTime))
```

Comment:
We're not storing total duration. How would this integrate with pushing metrics to the telemeter?

Reply:
The code here is only for the logging. This function can be the place where the duration metrics is populated. Storing the start time seemed enough as the total time can be deduced rather than stored.

Reply:
This PR should be created with future metrics collection in mind. We should try to create something that can either be extended or used by a client to achieve the metrics use case. If we  knowingly need to change this for metrics, we should try to address that now.